src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -188,14 +188,26 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
-    if (0 != net_mgmt(NET_REQUEST_WIFI_SCAN, mNetIf, NULL, 0))
+    if (!ssid.empty())
     {
-        ChipLogError(DeviceLayer, "Scan request failed");
-        return CHIP_ERROR_INTERNAL;
+        memcpy(mScanSsidBuffer, ssid.data(), ssid.size());
+        mScanParams.ssids[0] = reinterpret_cast<const char *>(&mScanSsidBuffer);
```

Comment:
I think SSID and scan params could be allocated on stack. Also, make sure you allocate enough buffer for null-terminator and validate the input string size:

```suggestion
        char ssidBuffer[DeviceLayer::Internal::kMaxWiFiSSIDLength + 1] = {};
        wifi_scan_params scanParams = {};
        
        VerifyOrReturnError(ssid.size() < sizeof(ssidBuffer), CHIP_ERROR_INVALID_ARGUMENT);
        memcpy(ssidBuffer, ssid.data(), ssid.size());
        scanParams.ssid[0] = ssidBuffer;
```

Reply:
The problem with those variables on the stac is that I think the function is finished before scanning (scanning might be done on the zephyr thead).
If I try directed scanning with your suggestions, the result is like without SSID given.

Reply:
After implementing suggested changes, I found some problems with directed scanning - one in few scan couldn't find network by ssid

Reply:
ok, nevermind then but please validate the SSID length in your solution and add `+1` to the buffer size :)

Reply:
Looks like the Wi-Fi driver uses only the shallow copy of `wifi_scan_params`, so we cannot rely on stack allocated objects here.

Reply:
Thanks for checking!